src/main/java/org/apache/sling/superimposing/SuperimposingManager.java

```diff
@@ -38,6 +54,6 @@
      *   Iterator is backed by a {@link java.util.concurrent.ConcurrentHashMap} and is safe to access
      *   even if superimposing resource providers are registered or unregistered at the same time.
      */
-    Iterator<SuperimposingResourceProvider> getRegisteredProviders();
+    Iterator<SuperimposingResourceProviderImpl> getRegisteredProviders();
```

Comment:
This seems incorrect - why are you now returning an implementation type from an interface? Didn't baselining complain?

Reply:
Collections4 IteratorUtils.unmodifiableIterator does not allow for implicit casting of as compared to Collections IteratorUtils.unmodifiableIterator. 
This is fixed now.